Mavros don't stop after SIGINT (^C)

Termination are broken in 0.7: mavros_node ignore SIGINT, but stops after SIGTERM.
Also now it ignore `rosnode kill mavros`.

Also termination at port closed event also broken (mavconn related).
```
[ERROR] [1408358771.110588398]: serial0:receive: End of file
terminate called after throwing an instance of 'std::system_error'
  what():  Resource deadlock avoided
```

--- a/libmavconn/src/serial.cpp
+++ b/libmavconn/src/serial.cpp
@@ -72,8 +72,8 @@ void MAVConnSerial::close() {
 	if (!is_open())
 		return;
 
-	serial_dev.close();
 	io_service.stop();
+	serial_dev.close();
 
 	// clear tx queue
 	for (auto &p : tx_q)

